### PR TITLE
Add link to endpoint context in middleware documentation

### LINKED ISSUE #3523: Document how to redirect from middleware
It was not easy to figure out the best way to redirect from middlware. Others have used the `Response.redirect` function which only supports full URLs. 

`context.redirect` is an easier approach since it supports relative paths and its also used in the middleware [example](https://github.com/withastro/astro/blob/main/examples/middleware/src/middleware.ts) found in withastro/astro.

Maybe we could document this, either in plain text or in a code example? Here's how I did it.

```ts
export const onRequest = defineMiddleware(async (context, next) => {
  if(shouldRedirect) {
   return context.redirect(`/relative/path`);
  }

  return next();
});
```

### PR DESCRIPTION
**Pull request by sasoria**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #3523
- This Pull request makes it easier to find the context API in the Middleware documentation. This is useful for finding a simple way to redirect, among other things.

### CODE REVIEW

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
I left you a suggestion, thanks for the PR @sasoria 😁

File: src/content/docs/en/guides/middleware.mdx [hunk 1]
```diff
@@ -197,7 +197,7 @@ An object that includes information to be made available to other middleware, AP
 
 This is an optional argument passed to `onRequest()` that may contain the [`locals`](#locals) object as well as any additional properties to be shared during rendering. For example, the `context` object may include cookies used in authentication.
 
-This is the same `context` object that is passed to API routes.
+This is the same [`context`](en/reference/api-reference/#endpoint-context) object that is passed to API routes.
```
Comment (dreyfus92) [inline src/content/docs/en/guides/middleware.mdx:200]:
```suggestion
This is the same [`context`](/en/reference/api-reference/#endpoint-context) object that is passed to API routes.
```

Comment (sasoria) [reply src/content/docs/en/guides/middleware.mdx]: Thanks!

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

**Review by ElianCodes — CHANGES_REQUESTED** [review]
Just a small nit

**Review by ElianCodes — APPROVED** [review]

**Comment by ElianCodes** [comment]
Thanks @sasoria ! Looks good now that `main` has been merged back into your PR!

Shipping it! ⛴️🚀

**Pull request merged**